src/test/unit/instruments.test.js

```diff
@@ -0,0 +1,123 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import * as JSZip from 'jszip';
+import * as fs from 'fs';
+import * as path from 'path';
+
+import { convertInstrumentsProfile } from '../../profile-logic/import/instruments';
+
+class MockFileSystemEntry {
+  isFile: boolean;
+  isDirectory: boolean;
+  name: string;
+  fullPath: string;
+  zip: typeof JSZip;
+  zipDir: any | null;
+  zipFile: JSZip.JSZipObject | null;
+
+  constructor(zip: typeof JSZip, fullPath: string) {
+    this.fullPath = fullPath;
+    this.zipFile = zip.file(fullPath);
+    this.isFile = !!this.zipFile;
+    this.zip = zip;
+
+    if (this.isFile) {
+      this.zipDir = null;
+      this.isDirectory = false;
+    } else {
+      this.zipDir = zip.folder(fullPath);
+      this.isDirectory = true;
+    }
+
+    this.name = path.basename(this.fullPath);
+  }
+
+  file(cb: (file: File) => void, errCb: (error: Error) => void) {
+    if (!this.zipFile) return errCb(new Error('Failed to extract file'));
+    this.zipFile
+      .async('blob')
+      .then(
+        blob => {
+          (blob: any).name = this.name;
+          cb(blob);
+        },
+        err => {
+          errCb(err);
+        }
+      )
+      .catch(errCb);
+
+    return undefined;
+  }
+
+  createReader() {
```

Comment:
@julienw , Can you please explain the first line? I didn't get you much.

Reply:
I meant that `createReader` would be present only if `zipDir` is not false, I mean in the real code, because this is only for directory (see the spec in https://developer.mozilla.org/en-US/docs/Web/API/FileSystemDirectoryEntry/createReader), not for other entries. So this mock is lying a bit here. I think it's good enough for our tests but please write a comment explaining this :)